### PR TITLE
Add support for retrieving carrier names

### PR DESCRIPTION
**Pull request by edwinnyawoli**

### CODE REVIEW

**Comment by emostar** [comment]
This is a great start, but it should have the matching iOS implementation. Would you be able to add that in @edwinnyawoli ?

**Comment by edwinnyawoli** [comment]
Implementing it in iOS is going to prove difficult because with Android there was [this](https://github.com/google/libphonenumber/tree/0dd0f034ff89d00f4d67cee8d8b88b27be4931bd/java/carrier) package that made it very simple and straightforward. I haven't seen anything like that in the [libPhoneNumber-iOS](https://github.com/iziz/libPhoneNumber-iOS) port or even in the cpp variant of the [libphonenumber](https://github.com/google/libphonenumber) project.

**Comment by emostar** [comment]
With that information, I agree. Can you just add a no-op, returning an empty string for iOS so the "getNameForNumber" method doesn't cause an exception to be raised?

**Comment by edwinnyawoli** [comment]
Updated pull request with no op ios implementation

**Comment by emostar** [comment]
@edwinnyawoli Sorry for dropping this, but I just came back and saw you added the no-op in. Thanks a lot for taking care of it!

I'll test this out and confirm everything is working, and then push a new version.

My biggest concern is having to keep this up to date, as the carrier library gets updated at least once a month with changes.

File: android/src/main/java/com/codeheadlabs/libphonenumber/LibphonenumberPlugin.java [hunk 1]
```diff
@@ -43,12 +46,26 @@ public void onMethodCall(MethodCall call, Result result) {
       case "formatAsYouType":
         formatAsYouType(call, result);
         break;
+      case "getNameForNumber":
+        handleGetNameForNumber(call, result);
```
Comment (emostar) [inline android/src/main/java/com/codeheadlabs/libphonenumber/LibphonenumberPlugin.java:50]: You are missing a `break` here

**Review by emostar — CHANGES_REQUESTED** [review]
Do you have a phone number that I can test with? My usual phone numbers are not returning any data for `getNameForNumber`.

We should also call it `getCaerrirNameForNumber` instead I think, as that is more explicit.

File: android/src/main/java/com/codeheadlabs/libphonenumber/LibphonenumberPlugin.java (hunk 1 above)
Comment (edwinnyawoli) [reply android/src/main/java/com/codeheadlabs/libphonenumber/LibphonenumberPlugin.java:50]: Oops...Yeah I missed that. I'll correct that and update the pull request.

**Pull request merged**